libraries/AP_Motors/AP_Motors_Class.cpp

```diff
@@ -162,14 +163,32 @@ void AP_Motors::rc_set_freq(uint32_t motor_mask, uint16_t freq_hz)
         hal.rcout->set_output_mode(mask, AP_HAL::RCOutput::MODE_PWM_DSHOT1200);
         break;
     case PWM_TYPE_PWM_RANGE:
+    case PWM_TYPE_PWM_ANGLE:
         /*
           this is a motor output type for multirotors which honours
-          the SERVOn_MIN/MAX values per channel
+          the SERVOn_MIN/MAX (and TRIM for angle) values per channel
+
+          Motor PWM min and max are hard coded to 1000 to 2000.
+          Range type offsets by 1000 to get 0 to 1000.
+          Angle type offsets by 1500 to get -500 to 500.
          */
-        _motor_pwm_range_mask |= motor_mask;
+
+        if (type == PWM_TYPE_PWM_RANGE) {
+            _motor_pwm_scaled.offset = 1000;
+        } else {
+            // PWM_TYPE_PWM_ANGLE
+            _motor_pwm_scaled.offset = 1500;
+        }
+
+        _motor_pwm_scaled.mask |= motor_mask;
         for (uint8_t i=0; i<16; i++) {
             if ((1U<<i) & motor_mask) {
-                SRV_Channels::set_range(SRV_Channels::get_motor_function(i), 1000);
+                if (type == PWM_TYPE_PWM_RANGE) {
+                    SRV_Channels::set_range(SRV_Channels::get_motor_function(i), 1000);
+                } else {
+                    // PWM_TYPE_PWM_ANGLE
```

Comment:
this feels odd. what is the expected range for `set_angle()`?

Reply:
You can set whatever you like. Here we just set the scale to match the range PWM input. We convert a 1000 to 2000 range to -500 to 500, so we set a angle of 500.

The whole range and angle thing is left over from before we did scaled values as floats, now we could just hard code everything to 1, but to maintain existing behavior we use random ranges, plane control surfaces use angle of 4500, plane throttle uses range of 100. Its all a bit odd.

Reply:
oh I see it now. If I read the code correctly, `SRV_Channels::set_angle()` should be called `SRV_Channels::set_maximum_angle()`

Reply:
which also explains why it is being used analogously to set_range()